### PR TITLE
Joint states tutorial

### PR DESCRIPTION
**Pull request by vatanaksoytezer**
Adds an rrbot model to demos and shows the usage of joint_states plugin.

### CODE REVIEW

File: ros_ign_gazebo_demos/launch/joint_states.launch.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            os.path.join(pkg_ros_ign_gazebo, 'launch', 'ign_gazebo.launch.py')),
+            launch_arguments={'ign_args': '-r empty.sdf'}.items(),
+    )
+
+    # RViz
+    rviz = Node(
+        package='rviz2',
+        executable='rviz2',
+        arguments=['-d', os.path.join(pkg_ros_ign_gazebo_demos, 'rviz', 'joint_states.rviz')],
+        parameters=[]
```
Comment (ahcorde) [inline ros_ign_gazebo_demos/launch/joint_states.launch.py]: remove

File: ros_ign_gazebo_demos/launch/joint_states.launch.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    )
+
+    # Spawn
+    rrbot_sdf_path =  os.path.join(pkg_ros_ign_gazebo_demos, 'models', 'rrbot.sdf')
+    # rrbot_sdf_path =  os.path.join(pkg_ros_ign_gazebo_demos, 'models', 'rrbot2.sdf')
```
Comment (ahcorde) [inline ros_ign_gazebo_demos/launch/joint_states.launch.py]: remove

File: ros_ign_gazebo_demos/launch/joint_states.launch.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    print(rrbot_sdf_path)
```
Comment (ahcorde) [inline ros_ign_gazebo_demos/launch/joint_states.launch.py]: remove

File: ros_ign_gazebo_demos/launch/joint_states.launch.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    spawn = Node(package='ros_ign_gazebo', executable='create',
+                arguments=[
+                    '-name', 'rrbot',
+                    '-file', rrbot_sdf_path,
+                    ],
+                output='screen',
+                )
```
Comment (ahcorde) [inline ros_ign_gazebo_demos/launch/joint_states.launch.py:52]: parentheses at the same indentation as the others

File: ros_ign_gazebo_demos/launch/joint_states.launch.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
+import os
+import xacro
```
Comment (ahcorde) [inline ros_ign_gazebo_demos/launch/joint_states.launch.py:2]: you are importing here xacro, but you are not using it

**Review by ahcorde — CHANGES_REQUESTED** [review]
You are adding here the URDF and the SDF and using it both in the launch file. I don't like this approach because if you modify one you must need to modify the other one. You should use only the URDF, with xacro (which is imported but not used) you can convert the URDF and uing it to spawn the model. In this [file](https://github.com/ros-simulation/gazebo_ros_demos/blob/a5b8877fc8398e044e4aedccf69afb6c39ee298b/rrbot_gazebo/launch/rrbot_world.launch.py) you can find how to do it.

Another question that I open here it's that the `rrbot.urdf` should live only in one repository we should not copy this file in every repo that make use of it.

**Review by ahcorde — CHANGES_REQUESTED** [review]
You should also sign the [PR](https://github.com/ignitionrobotics/ros_ign/pull/156/checks?check_run_id=2685739378)

**Comment by ahcorde** [comment]
@vatanaksoytezer, I have seen in your profile that you are working for Picknik maybe we can speed up this other PR https://github.com/ros-simulation/gazebo_ros_demos/pull/41 where I think the rrobot description should live.

**Comment by vatanaksoytezer** [comment]
> @vatanaksoytezer, I have seen in your profile that you are working for Picknik maybe we can speed up this other PR [ros-simulation/gazebo_ros_demos#41](https://github.com/ros-simulation/gazebo_ros_demos/pull/41) where I think the rrobot description should live.

Sure I will ping Dave internally.

File: ros_ign_gazebo_demos/launch/joint_states.launch.py (hunk 2 above)
Comment (vatanaksoytezer) [reply ros_ign_gazebo_demos/launch/joint_states.launch.py]: Oops forgot those in cleanup

**Comment by vatanaksoytezer** [comment]
> Another question that I open here it's that the rrbot.urdf should live only in one repository we should not copy this file in every repo that make use of it.

I would agree that but I really wouldn't want to make `ign_gazebo_demos` depend to `gazebo_ros_demos` for rrbot only. Maybe you guys can consider opening up a rrbot description and depend on that?

**Comment by vatanaksoytezer** [comment]
> You are adding here the URDF and the SDF and using it both in the launch file. I don't like this approach because if you modify one you must need to modify the other one. You should use only the URDF, with xacro (which is imported but not used) you can convert the URDF and uing it to spawn the model. In this [file](https://github.com/ros-simulation/gazebo_ros_demos/blob/a5b8877fc8398e044e4aedccf69afb6c39ee298b/rrbot_gazebo/launch/rrbot_world.launch.py) you can find how to do it.

@ahcorde Follow up question on this one. Since this is to demonstrate the usage of `joint_states` plugin in ignition, where should I put the plugin if I don't have the sdf file? I was mostly looking up for @chapulina 's `dolly` package where she also defines the robot from  an sdf file.

**Comment by vatanaksoytezer** [comment]
Another problem that I just realised is if rrbot_description live somewhere else and contains `gazebo` and `ros2_control` stuff that is not compatible with ignition (yet), wouldn't it be problematic?

**Comment by vatanaksoytezer** [comment]
@ahcorde update: 

- Pinged Dave on gazebo PR.
- Signed off the commits and cleaned up stuff, except for the sdf (see my comments above)

**Comment by vatanaksoytezer** [comment]
Ignore the comments above on sdf, just figured that out. I'll fix in and send updates soon

**Comment by vatanaksoytezer** [comment]
This should work, but the colors (and I think frictions) are not parsed correctly to ign. If you point me a way I can fix it right away

**Comment by ahcorde** [comment]
@vatanaksoytezer, I can see the color in ignition 

Selección_104

**Review by ahcorde — APPROVED** [review]
LGTM

**Pull request merged**